ui/pages/swaps/prepare-swap-page/view-quote-price-difference.js

```diff
@@ -0,0 +1,107 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+
+import { I18nContext } from '../../../contexts/i18n';
+import Box from '../../../components/ui/box';
+import {
+  DISPLAY,
+  AlignItems,
+  SEVERITIES,
+  Size,
+  TextVariant,
+  BLOCK_SIZES,
+} from '../../../helpers/constants/design-system';
+import { GasRecommendations } from '../../../../shared/constants/gas';
+import {
+  BannerAlert,
+  Text,
+  ButtonLink,
+} from '../../../components/component-library';
+
+export default function ViewQuotePriceDifference(props) {
+  const {
+    usedQuote,
+    sourceTokenValue,
+    destinationTokenValue,
+    onAcknowledgementClick,
+    acknowledged,
+    priceSlippageFromSource,
+    priceSlippageFromDestination,
+    priceDifferencePercentage,
+    priceSlippageUnknownFiatValue,
+  } = props;
+
+  const t = useContext(I18nContext);
+
+  let priceDifferenceTitle = '';
+  let priceDifferenceMessage = '';
+  let priceDifferenceClass = '';
+  if (priceSlippageUnknownFiatValue) {
+    // A calculation error signals we cannot determine dollar value
+    priceDifferenceTitle = t('swapPriceUnavailableTitle');
+    priceDifferenceMessage = t('swapPriceUnavailableDescription');
+    priceDifferenceClass = GasRecommendations.high;
```

Comment:
Can these first values be assigned at the time of `let`, and then replaced with if `! priceSlippageUnknownFiatValue`.